discovery.js

```
@@ -41,23 +44,26 @@ DiscoveryClient.prototype._update = function (update) {
 };
 
 /* Connect to the Discovery servers given the endpoint of any one of them */
-DiscoveryClient.prototype.connect = function (onComplete) {
+DiscoveryClient.prototype.connect = function (cb) {
   var disco = this;
   request({
     url: "http://" + this.host + "/watch",
     json: true
   }, function (error, response, update) {
     if (error) {
-      onComplete(error);
+      console.error('Discovery Service::Connect:: error: %s', error);
+      cb(error);
       return;
     }
-    if (response.statusCode != 200) {
-      onComplete(new Error("Unable to initiate discovery: " + JSON.stringify(update)), undefined, undefined);
+    if (response.statusCode !== 200) {
+      disco.logger.log('error', 'Discovery Service::Connect:: Response Code is not 200: ' + JSON.stringify(update));
+      cb("Unable to initiate discovery: " + JSON.stringify(update));
       return;
     }
 
     if (!update.fullUpdate) {
-      onComplete(new Error("Expecting a full update: " + JSON.stringify(update)), undefined, undefined);
+      console.error('Discovery Service::Connect:: fullUpdate is undefined %s', JSON.stringify(update));
+      cb("Expecting a full update: " + JSON.stringify(update));
```

I'm not sure if we want to throw an error it will break the app unless we use node domain to wrap it.

from the other hand we can't really start the app without discovery...

@stevenschlansker any opinion?